backend/src/main/java/com/team4/backend/service/EmailSenderService.java

```diff
@@ -0,0 +1,56 @@
+package com.team4.backend.service;
+
+import com.team4.backend.exception.UserNotFoundException;
+import lombok.extern.java.Log;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Log
+@Service
+public class EmailSenderService {
+
+
+    private final StudentService studentService;
+
+    private final InternshipOfferService internshipOfferService;
+
+    private final JavaMailSender javaMailSender;
+
+    public EmailSenderService(StudentService studentService, InternshipOfferService internshipOfferService, JavaMailSender javaMailSender) {
+        this.studentService = studentService;
+        this.internshipOfferService = internshipOfferService;
+        this.javaMailSender = javaMailSender;
+    }
+
+    public Mono<Void> sendEmailToStudent(String sender, String receiver, String subject, String content) {
+        return studentService.findByEmail(receiver)
+                .flatMap(student -> internshipOfferService.monitorOffersInterestedStudentsContainsStudentEmail(receiver, sender))
+                .flatMap( studentInMonitorListResponse ->
+                        studentInMonitorListResponse ?
+                            sendEmail(sender, receiver, subject, content) :
+                            Mono.error(new UserNotFoundException("Given receiver email does not correspond to student that applied to monitor's offer")));
```

Comment:
Use if instead of ternary operator for readability

Reply:
👍